apps/chat/views.py

```diff
@@ -0,0 +1,17 @@
+from django.contrib.auth.decorators import permission_required
+from django.http import FileResponse, Http404
+from django.shortcuts import get_object_or_404
+
+from apps.files.models import File
+from apps.teams.decorators import login_and_team_required
+
+
+@login_and_team_required
+@permission_required("assistants.view_threadtoolresources")
+def download_file(request, team_slug: str, pk: int):
```

Comment:
There is already a view for this: https://github.com/dimagi/open-chat-studio/blob/bb9bdf14065245e475fefbc2858c3a02ab78ec8f/apps/files/urls.py#L8

If you want an additional permission you could extend that view and add the decorator

Reply:
Another option here is to link to the `ThreadToolResources` object which would allow us to validate the user's access as well. I think we'll need to apply the same checks as in `apps.experiments.views.experiment.experiment_review` using the `@verify_session_access_cookie` decorator (or we just don't allow unauthenticated users to download the files).

I think it also makes sense to move this to the `experiments` views file so that it can be alongside the other chat views.

Reply:
let me know what you think about [this approach](https://github.com/dimagi/open-chat-studio/pull/525/commits/865f469133b090f8dd5fe91ef544ff2eb46b8638)